src/app/jscaip/Coord.ts

```diff
@@ -35,11 +35,8 @@ export class Coord extends Vector {
         super(x, y);
     }
     public getNext(dir: Vector, distance?: number): Coord {
-        // return the next coord in the direction 'dir'
-        distance = distance == null ? 1 : distance;
-        const newX: number = this.x + (distance * dir.x);
-        const newY: number = this.y + (distance * dir.y);
-        return new Coord(newX, newY);
+        const combinedVector: Vector = this.combine(dir, distance);
```

Comment:
faut changer le type de combine pour qu'il retourne pas un Vector mais un this !